### PR TITLE
Revise TRAC-876

### PR DESCRIPTION
**Pull request by cdeaneGit**
Please test if I correctly added field 'utk_mods_etd_author_orcid_s' to solr.
Code in slurp_all_MODS_to_solr.xslt

Test what happens if you add an orcid on the form.
Test what happens if you do not add an orcid on the form.

@mark @bridger

### CODE REVIEW

**Comment by markpbaggett** [comment]
@cdeaneGit, testing now.

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt [hunk 1]
```diff
@@ -57,6 +57,17 @@
       </xsl:otherwise>
         </xsl:choose>
     </field>
+
+    <-- JIRA TRAC-876 Define utk_mods_etd_author_orcid in Solr -->
+
+    <xsl:if test="mods:name[@type='orcid']">
```
Comment (markpbaggett) [inline fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: This isn't working.  One problem here is that you're reading the wrong attribute value.  You should be reading mods:name[@authority='orcid'].

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt [hunk 2]
```diff
@@ -57,6 +57,17 @@
       </xsl:otherwise>
         </xsl:choose>
     </field>
+
+    <-- JIRA TRAC-876 Define utk_mods_etd_author_orcid in Solr -->
```
Comment (markpbaggett) [inline fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: This comment is missing a bang.

Comment (CanOfBees) [inline fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Looks like there's some invalid syntax here; comments should be `<!-- $text -->`

**Review by CanOfBees — CHANGES_REQUESTED** [review]

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt (hunk 1 above)
Comment (markpbaggett) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Your conditional needs to start with this:

<xsl:if test="/mods:mods/mods:name[@authority='orcid']">

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt [hunk 3]
```diff
@@ -57,6 +57,17 @@
...
+
+    <xsl:if test="mods:name[@type='orcid']">
+        <xsl:variable name="orcidtrim" select="substring-after(mods:name[@valueURI],'http://orcid.org/')"/>
```
Comment (markpbaggett) [inline fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Your variable match is wrong:

<xsl:variable name="orcidtrim" select="substring-after(/mods:mods/mods:name/@valueURI,'http://orcid.org/')"/>

**Review by markpbaggett — CHANGES_REQUESTED** [review]

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt (hunk 2 above)
Comment (cdeaneGit) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Deleted comment.

Comment (cdeaneGit) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: My bad.

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt (hunk 1 above)
Comment (cdeaneGit) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Revised test statement

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt (hunk 3 above)
Comment (cdeaneGit) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Revised to correct path to /mods:mods/mods:name[@valueURI]

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt (hunk 2 above)
Comment (cdeaneGit) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Deleted.

**Comment by CanOfBees** [comment]
@cdeaneGit double-checking this now. During a review yesterday evening I thought I saw a weird issue, so I'm tearing down the VM and rebuilding.

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt [hunk 4]
```diff
@@ -58,10 +59,9 @@
         </xsl:choose>
     </field>
 
-    <-- JIRA TRAC-876 Define utk_mods_etd_author_orcid in Solr -->
 
-    <xsl:if test="mods:name[@type='orcid']">
-        <xsl:variable name="orcidtrim" select="substring-after(mods:name[@valueURI],'http://orcid.org/')"/>
+    <xsl:if test="/mods:mods/mods:name[@authority='orcid']">
+        <xsl:variable name="orcidtrim" select="substring-after(/mods:mods/mods:name[@valueURI],'http://orcid.org/')"/>
```
Comment (CanOfBees) [inline fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: There's still an issue with the xsl:variable's select instruction: in the first argument of the substring-after() function, the actual value being selected is a `/mods:mods/mods:name` with an `@valueURI`, when really you're wanting `/mods:mods/mods:name/@valueURI`.  Here's a working example:

```xml
<xsl:if test="@authority='orcid'">
  <xsl:variable name="vORCID" select="substring-after(@valueURI, 'http://orcid.org/')"/>
  <field name="utk_mods_etd_author_orcid_s">
    <xsl:value-of select="$vORCID"/>
  </field>   
</xsl:if>
```
This is a slight modification on @markpbaggett's suggestion from yesterday. This keeps the current context (i.e. the `xsl:template match="..."` that is currently being used). 


chd:  I revised per  Bridger's suggestions here.

**Review by CanOfBees — CHANGES_REQUESTED** [review]

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt (hunk 4 above)
Comment (markpbaggett) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: @CanOfBees, sorry.  I meant to copy and paste that in.  I just missed that part in my review.  You're absolutely correct.

**Review by CanOfBees — APPROVED** [review]

**Review by markpbaggett — APPROVED** [review]

**Pull request merged**